src/util.h

```diff
@@ -571,5 +571,56 @@ std::string json_escape_utf8(const std::string& val, bool escape_printable_contr
 std::string json_escape_utf8(const char* val, size_t val_size,
                              bool escape_printable_controls = true);
 
+/**
+ * Splits a string at all occurrences of a delimiter. Successive occurrences
+ * of the delimiter will be split into multiple pieces.
+ *
+ * \note This function is not UTF8-aware.
+ */
+template <typename T> std::vector<T> split(T s, const T& delim)
```

Comment:
Could this be `template <typename T> std::vector<T> split(const T s, const T& delim)`, to allow for const string arguments?

Reply:
It'd require some changes since `s` gets modified by the loop internally, but certainly doable.

Reply:
Ah, I missed the substring operation assigning to the same variable. That's fine then.